### PR TITLE
Add CI check to run terraform plan

### PR DESCRIPTION
**Pull request by nickclyde**
Resolves https://app.clickup.com/t/3cq8zhu

### CODE REVIEW

**Comment by celestemespinoza** [comment]
Task linked: [CU-3cq8zhu Add CI check to run terraform plan](https://app.clickup.com/t/3cq8zhu)

**Review by DanPaseltiner — APPROVED** [review]
LGTM!

**Review by BradySkylight — APPROVED** [review]
Looks good to me as well.

**Comment by nickclyde** [comment]
I decided to go rogue and completely change this PR after getting approvals 😆 

I think it makes sense to split the terraform checks into a separate workflow that only triggers is any file changes in the `terraform/` directory. That way PRs that only touch python code don't have to wait for a terraform plan to run. Let me know what you think about this!

File: .github/workflows/terraformChecks.yaml
Comment (DanPaseltiner) [inline .github/workflows/terraformChecks.yaml:103]: Where is our value for `env.ENVIRONMENT` coming from?

**Review by DanPaseltiner — COMMENTED** [review]
I think this makes sense to me. In general I really like the approach of making our CI/CD workflows more targeted so we aren't running irrelevant workflows on every PR. One quick question in a comment below.

File: .github/workflows/terraformChecks.yaml
Comment (nickclyde) [reply .github/workflows/terraformChecks.yaml:103]: That gets set on the step on line 89. It will default to dev, but can be dispatched with a different environment as input... once I add that to the top of the file 😅

Comment (DanPaseltiner) [reply .github/workflows/terraformChecks.yaml:108]: Whoops! Totally missed that. Makes sense to me.

**Review by DanPaseltiner — APPROVED** [review]
Thanks for reworking this!

**Pull request merged**
